smart_open/tests/test_ftp.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+
+import logging
+import unittest
+from unittest.mock import patch, call
+
+import smart_open
+from smart_open import ftp
+
+
+class FTPOpen(unittest.TestCase):
+    @patch("smart_open.ftp.FTP")
```

Comment:
Honestly I'm not sure what these tests really give us. All it is testing is that we are calling certain methods from the ftplib. I'd suggest:

1. Running an FTP server in CI and writing integration tests against it
2. Implementing mocks that imitate functionality of a FTP server. There's some good examples of us doing this [here for GCS](https://github.com/RaRe-Technologies/smart_open/blob/4c6bc38d6b35aa75e5e9b8041bb80f0f956ac486/smart_open/tests/test_gcs.py#L46)